fluent/protocol/transport.go

```diff
@@ -139,103 +137,27 @@ type Entry struct {
 
 type MessageOptions struct {
 	Size       int    `msg:"size"`
-	Chunk      string `msg:"chunk"`
-	Compressed string `msg:"compressed"`
-}
-
-// Message is used to send a single event at a time
-//msgp:tuple Message
-type Message struct {
-	// Tag is a dot-delimited string used to categorize events
-	Tag       string
-	Timestamp int64
-	Record    Record
-	// Options - used to control server behavior.
-	Options *MessageOptions
-}
-
-// MessageExt
-//msgp:tuple MessageExt
-type MessageExt struct {
-	Tag       string
-	Timestamp EventTime `msg:"eventTime,extension"`
-	Record    Record
-	Options   *MessageOptions
-}
-
-// PackedForwardMessage is just like ForwardMessage, except that the events
-// are carried as a msgpack binary stream
-//msgp:tuple PackedForwardMessage
-type PackedForwardMessage struct {
-	// Tag is a dot-delimited string used to categorize events
-	Tag string
-	// EventStream is the set of events (entries in Fluent-speak) serialized
-	// into a msgpack byte stream
-	EventStream []byte
-	// Options - used to control server behavior.  Same as above, may need to
-	// switch to interface{} or similar at some point.
-	Options *MessageOptions
-}
-
-// NewPackedForwardMessage creates a PackedForwardMessage from the supplied
-// tag, EntryList, and MessageOptions.  Regardless of the options supplied,
-// this function will set opts[OPT_SIZE] to the length of the entry list.
-func NewPackedForwardMessage(
-	tag string,
-	entries EntryList,
-	opts *MessageOptions,
-) *PackedForwardMessage {
-	// set the options size to be the number of entries
-	opts.Size = len(entries)
-
-	msg := &PackedForwardMessage{
-		Tag:         tag,
-		EventStream: eventStream(entries),
-		Options:     opts,
-	}
-	return msg
-}
-
-func eventStream(entries EntryList) []byte {
-	var buf bytes.Buffer
-	w := msgp.NewWriter(&buf)
-	for _, e := range entries {
-		// TODO: capture and return error
-		_ = e.EncodeMsg(w)
-	}
-	w.Flush()
-
-	return buf.Bytes()
-}
-
-// CompressedPackedForwardMode is just like PackedForwardMessage, except that
-// the msgpack byte stream containing the events/entries is compressed using
-// gzip.  The protocol spec states that the event stream may be formed by
-// concatenating multiple gzipped binary strings, but we do not claim to
-// support that yet.
-// Users of this type MUST set the option "compressed" => "gzip"
-//msgp:tuple CompressedPackedForwardMessage
-type CompressedPackedForwardMessage struct {
```

Comment:
We don't need CompressedPackedForwardMessage as it's just a PackedForwardMessage where the events are gzipped.